Emdash terminal environment variables break external CLI tools (e.g: cmake fails in worktree terminal but works in normal terminal)

## Bug Description

When opening a terminal from a task in Emdash (AppImage version), the terminal environment inherits Emdash-specific internal variables that interfere with external CLI tools. This causes tools like `cmake` to fail when executed from the Emdash terminal, while the same tools work correctly when:
- Run from a normal system terminal outside Emdash
- Executed directly by Auggie within Emdash (without opening a terminal)

This suggests that Emdash's terminal spawning process is polluting the environment with AppImage-specific variables that should remain internal to the Emdash process.

## Context

- Related to #449 (AppImage PATH environment issues)
- Related to #485 (Python/PYTHONHOME environment conflicts)
- Related to #595 (Terminal CWD issues with AppImage)

## Environment

| Component | Version |
|-----------|--------|
| **Operating System** | Linux Mint 22 (Kernel 6.8.0-90-generic) |
| **Hardware** | ThinkPad X1 Extreme Gen 4i |
| **Node.js** | v22.14.0 |
| **npm** | 11.6.2 |
| **Emdash** | v0.3.50 (AppImage) |
| **AppImage Location** | `/home/levi/Packages/EMDash/emdash-x86_64.AppImage` |
| **cmake** | System-installed via package manager |

## Steps to Reproduce
(following workaround detailed at #595 ) 

1. Launch Emdash using the AppImage (`emdash-x86_64.AppImage`)
2. Open or create a task (which creates a worktree)
3. Right-click on the task and select "Open in" → "Terminal" (follow #595 ) 
4. In the opened terminal, run: `env | grep emdash` (to see Emdash/AppImage variables)
5. Attempt to run `cmake`
6. **Observe**: cmake fails with "Could not find CMAKE_ROOT" error
7. **Compare**: Open a normal system terminal and run `cmake` (works correctly)
8. **Compare**: Run cmake via Auggie directly in Emdash (works correctly)

## Expected Behavior

- Emdash terminal should provide a clean environment similar to a normal system terminal
- External CLI tools should work the same way in Emdash terminals as in system terminals
- AppImage-specific internal variables should not leak into the user's terminal environment

## Actual Behavior

The Emdash terminal contains AppImage-specific environment variables that pollute the user's environment:

### Polluted Environment Variables

```bash
$ env | grep emdash
PATH=...:/tmp/.mount_emdashw4Kk4P:/tmp/.mount_emdashw4Kk4P/usr/sbin:...
LD_LIBRARY_PATH=/tmp/.mount_emdashw4Kk4P/usr/lib::/usr/local/cuda/lib64:...
APPIMAGE=/home/levi/Packages/EMDash/emdash-x86_64.AppImage
APPDIR=/tmp/.mount_emdashw4Kk4P
XDG_DATA_DIRS=/tmp/.mount_emdashw4Kk4P/usr/share/:...
CHROME_DESKTOP=emdash.desktop
GSETTINGS_SCHEMA_DIR=/tmp/.mount_emdashw4Kk4P/usr/share/glib-2.0/schemas:
```

### cmake Failure

```
$ cmake
CMake Error: Could not find CMAKE_ROOT !!!
CMake has most likely not been installed correctly.
Modules directory not found in

Usage
  cmake [options] <path-to-source>
  ...

Run 'cmake --help' for more information.
```

## Related Issues

- #449 - AppImage PATH environment issues
- #485 - Python/PYTHONHOME environment conflicts  
- #595 - Terminal CWD issues with AppImage

----


**[UPDATE - 2026-01-13 00:45 UTC]**

- Same observed behavior occurs when opening `"Open in" → "VSCode" (following #595 )`
- The IDE opens correctly and on the correct worktree workspace, but the VSCode internal Terminal gets also polluted with emdash env variables. 
- However, interestingly enough, opening an external terminal _from_ the polluted VSCode, generates a clean terminal that behaves as expected

--- a/src/main/ipc/appIpc.ts
+++ b/src/main/ipc/appIpc.ts
@@ -6,6 +6,7 @@ import { ensureProjectPrepared } from '../services/ProjectPrep';
 import { getAppSettings } from '../settings';
 import { getAppById, OPEN_IN_APPS, type OpenInAppId, type PlatformKey } from '@shared/openInApps';
 import { databaseService } from '../services/DatabaseService';
+import { buildExternalToolEnv } from '../utils/childProcessEnv';
 
 const UNKNOWN_VERSION = 'unknown';
 
@@ -21,7 +22,11 @@ const execCommand = (
   return new Promise((resolve, reject) => {
     exec(
       command,
-      { maxBuffer: opts?.maxBuffer ?? 8 * 1024 * 1024, timeout: opts?.timeout ?? 30000 },
+      {
+        maxBuffer: opts?.maxBuffer ?? 8 * 1024 * 1024,
+        timeout: opts?.timeout ?? 30000,
+        env: buildExternalToolEnv(),
+      },
       (error, stdout) => {
         if (error) return reject(error);
         resolve(stdout ?? '');
@@ -226,7 +231,7 @@ export function registerAppIpc() {
               const terminalCommand = `osascript -e 'tell application "Terminal" to do script "${escapedCommand}"' -e 'tell application "Terminal" to activate'`;
 
               await new Promise<void>((resolve, reject) => {
-                exec(terminalCommand, (err) => {
+                exec(terminalCommand, { env: buildExternalToolEnv() }, (err) => {
                   if (err) return reject(err);
                   resolve();
                 });
@@ -239,7 +244,7 @@ export function registerAppIpc() {
               const terminalCommand = `osascript -e 'tell application "iTerm" to create window with default profile command "${escapedCommand}"'`;
 
               await new Promise<void>((resolve, reject) => {
-                exec(terminalCommand, (err) => {
+                exec(terminalCommand, { env: buildExternalToolEnv() }, (err) => {
                   if (err) return reject(err);
                   resolve();
                 });
@@ -259,7 +264,7 @@ export function registerAppIpc() {
               const terminalCommand = `ghostty -e ${quoted(sshCommand)}`;
 
               await new Promise<void>((resolve, reject) => {
-                exec(terminalCommand, (err) => {
+                exec(terminalCommand, { env: buildExternalToolEnv() }, (err) => {
                   if (err) return reject(err);
                   resolve();
                 });
@@ -328,7 +333,7 @@ export function registerAppIpc() {
         }
 
         await new Promise<void>((resolve, reject) => {
-          exec(command, (err) => {
+          exec(command, { env: buildExternalToolEnv() }, (err) => {
             if (err) return reject(err);
             resolve();
           });
@@ -349,7 +354,7 @@ export function registerAppIpc() {
     // Helper to check if a command exists
     const checkCommand = (cmd: string): Promise<boolean> => {
       return new Promise((resolve) => {
-        exec(`command -v ${cmd} >/dev/null 2>&1`, (error) => {
+        exec(`command -v ${cmd} >/dev/null 2>&1`, { env: buildExternalToolEnv() }, (error) => {
           resolve(!error);
         });
       });
@@ -358,18 +363,26 @@ export function registerAppIpc() {
     // Helper to check if macOS app exists by bundle ID
     const checkMacApp = (bundleId: string): Promise<boolean> => {
       return new Promise((resolve) => {
-        exec(`mdfind "kMDItemCFBundleIdentifier == '${bundleId}'"`, (error, stdout) => {
-          resolve(!error && stdout.trim().length > 0);
-        });
+        exec(
+          `mdfind "kMDItemCFBundleIdentifier == '${bundleId}'"`,
+          { env: buildExternalToolEnv() },
+          (error, stdout) => {
+            resolve(!error && stdout.trim().length > 0);
+          }
+        );
       });
     };
 
     // Helper to check if macOS app exists by name
     const checkMacAppByName = (appName: string): Promise<boolean> => {
       return new Promise((resolve) => {
-        exec(`osascript -e 'id of application "${appName}"' 2>/dev/null`, (error) => {
-          resolve(!error);
-        });
+        exec(
+          `osascript -e 'id of application "${appName}"' 2>/dev/null`,
+          { env: buildExternalToolEnv() },
+          (error) => {
+            resolve(!error);
+          }
+        );
       });
     };
 

--- a/src/main/utils/__tests__/childProcessEnv.test.ts
+++ b/src/main/utils/__tests__/childProcessEnv.test.ts
@@ -1,0 +1,55 @@
+import { describe, expect, it } from 'vitest';
+import { buildExternalToolEnv } from '../childProcessEnv';
+
+describe('buildExternalToolEnv', () => {
+  it('removes AppImage-only keys and strips mount paths from PATH-like vars', () => {
+    const env = buildExternalToolEnv({
+      APPDIR: '/tmp/.mount_emdashAbCd',
+      APPIMAGE: '/home/user/emdash.AppImage',
+      ARGV0: 'AppRun',
+      CHROME_DESKTOP: 'emdash.desktop',
+      GSETTINGS_SCHEMA_DIR: '/tmp/.mount_emdashAbCd/usr/share/glib-2.0/schemas',
+      OWD: '/tmp',
+      PATH: '/usr/local/bin:/tmp/.mount_emdashAbCd/usr/bin:/usr/bin',
+      LD_LIBRARY_PATH: '/tmp/.mount_emdashAbCd/usr/lib:/usr/local/cuda/lib64',
+      XDG_DATA_DIRS: '/tmp/.mount_emdashAbCd/usr/share:/usr/share',
+      HOME: '/home/user',
+      USER: 'user',
+      KEEP_ME: 'yes',
+    });
+
+    expect(env.APPDIR).toBeUndefined();
+    expect(env.APPIMAGE).toBeUndefined();
+    expect(env.ARGV0).toBeUndefined();
+    expect(env.CHROME_DESKTOP).toBeUndefined();
+    expect(env.GSETTINGS_SCHEMA_DIR).toBeUndefined();
+    expect(env.OWD).toBeUndefined();
+
+    expect(env.PATH).toBe('/usr/local/bin:/usr/bin');
+    expect(env.LD_LIBRARY_PATH).toBe('/usr/local/cuda/lib64');
+    expect(env.XDG_DATA_DIRS).toBe('/usr/share');
+
+    expect(env.HOME).toBe('/home/user');
+    expect(env.USER).toBe('user');
+    expect(env.KEEP_ME).toBe('yes');
+  });
+
+  it('removes Python vars only when they point into AppImage mount paths', () => {
+    const stripped = buildExternalToolEnv({
+      APPDIR: '/tmp/.mount_emdashZZ',
+      PYTHONHOME: '/tmp/.mount_emdashZZ/usr',
+      PYTHONPATH: '/tmp/.mount_emdashZZ/usr/lib/python3.11',
+    });
+
+    expect(stripped.PYTHONHOME).toBeUndefined();
+    expect(stripped.PYTHONPATH).toBeUndefined();
+
+    const kept = buildExternalToolEnv({
+      PYTHONHOME: '/opt/python',
+      PYTHONPATH: '/opt/python/lib',
+    });
+
+    expect(kept.PYTHONHOME).toBe('/opt/python');
+    expect(kept.PYTHONPATH).toBe('/opt/python/lib');
+  });
+});

--- a/src/main/utils/childProcessEnv.ts
+++ b/src/main/utils/childProcessEnv.ts
@@ -1,0 +1,51 @@
+const APPIMAGE_ENV_KEYS = [
+  'APPDIR',
+  'APPIMAGE',
+  'ARGV0',
+  'CHROME_DESKTOP',
+  'GSETTINGS_SCHEMA_DIR',
+  'OWD',
+] as const;
+
+const APPIMAGE_PATH_LIKE_ENV_KEYS = ['PATH', 'LD_LIBRARY_PATH', 'XDG_DATA_DIRS'] as const;
+
+function stripPathLikeAppImageEntries(value: string, appDir?: string): string {
+  const separator = process.platform === 'win32' ? ';' : ':';
+  const parts = value.split(separator).filter(Boolean);
+  if (parts.length === 0) return value;
+
+  const filtered = parts.filter((part) => {
+    if (appDir && part.startsWith(appDir)) return false;
+    if (part.includes('/tmp/.mount_')) return false;
+    return true;
+  });
+
+  return filtered.join(separator);
+}
+
+export function buildExternalToolEnv(baseEnv: NodeJS.ProcessEnv = process.env): NodeJS.ProcessEnv {
+  const env: NodeJS.ProcessEnv = { ...baseEnv };
+  const appDir = typeof baseEnv.APPDIR === 'string' ? baseEnv.APPDIR : undefined;
+
+  for (const key of APPIMAGE_ENV_KEYS) {
+    delete env[key];
+  }
+
+  for (const key of APPIMAGE_PATH_LIKE_ENV_KEYS) {
+    const value = env[key];
+    if (typeof value !== 'string' || value.length === 0) continue;
+    const cleaned = stripPathLikeAppImageEntries(value, appDir);
+    if (cleaned.length > 0) env[key] = cleaned;
+    else delete env[key];
+  }
+
+  for (const key of ['PYTHONHOME', 'PYTHONPATH'] as const) {
+    const value = env[key];
+    if (typeof value !== 'string' || value.length === 0) continue;
+    if ((appDir && value.startsWith(appDir)) || value.includes('/tmp/.mount_')) {
+      delete env[key];
+    }
+  }
+
+  return env;
+}

